app/models/spree/order_decorator.rb

```diff
@@ -0,0 +1,25 @@
+Spree::Order.class_eval do
+
+  state_machine.after_transition to: :complete, do: :update_multi_coupon_codes
+
+  def update_multi_coupon_codes
+    promotion_ids = all_adjustments.promotion.eligible.map { |a| a.source.promotion_id }
+    promotion = promotions.where(id: promotion_ids).where(multi_coupon: true).last
```

Comment:
can be replaced with?

promotions.find_by(id: promotion_ids, multi_coupon: true)